### ISSUE TITLE
Showing core dev support for a (feature) request. Please don't use emoticons for this

### ISSUE THREAD

**Issue by paramat** [minetest/minetest#7861]
Since we have hundreds of feature requests and are likely to have ever-increasing numbers, i feel we need some help with managing these, so i have added a new label 'Supported by core dev' (name could be changed) which i suggest is added when a core dev supports a feature request, or when core dev support is noticed in a feature request.

This will save having to read through the whole thread repeatedly every few months to see if the feature request is supported or not. Due to making a big effort to manage issues i know this will save a lot of time and effort.

Importantly i request that core devs do not use the emoticons in the first post to show their official support, for several reasons:
* It's an unsuitable place to register official support, emoticons are casual playful things mainly used by non-core-devs.
* It's a non-obvious place to look for core dev support. I have only just started to hover over the thumbs-up to see the list and scan it for a core dev (which is a pain to do, tiny font). Until now i never thought of doing this because it seemed obvious that was the most unsuitable place to register official support.
* It's an unsafe way to store official support, as the list of names of those who thumbed-up is low priority for GitHub to preserve, this listing is highly likely to be removed or cleared and reset.
* The listing is incomplete once a certain number of people have thumbed-up, it says: "blah, blah and 5 other people". So core dev support can easily become unviewable later.

I suggest core devs add a comment with a +1 or :+1: to clearly show their support, just like approving a PR, and add the label if necessary.

//////////////////////////////////
@rubenwardy 
I saw your discussion in IRC http://irc.minetest.net/minetest-hub/2018-11-12#i_5438095
~I suggest that we could possibly also have a label for '2+ core devs supporting' if you want to tie this in to concept approval.~

To answer your question, i think a feature request should have 2+ core devs supporting to be considered an approved concept.

"in terms of implementation: MinetestBot maintains a label called "Concept approval needed". To approve an issues (inc: PRs) two core devs need to do :+1: on the reactions"
If by 'reactions' you mean the emoticons in the first post, that's not a good idea as explained above.

**Comment by paramat** [comment]
It may be the easiest and user-friendly way to show support, but it's also the worst way as explained. 

If a bot counts reactions, how does it count a reaction lost because the list is large (4th point)? Would it check regularly to not miss a reaction? If so this would also address the 3rd point.

So counting reactions may indeed not be an issue for the Bot itself. However, consider that you would then have to encourage core devs to use reactions to express approvals, which is a very bad idea for the reasons above. Approvals are far too important to trust to an emoticon click-list.

I'm wondering why do approvals need to be automated? Core devs already have to read and process issues and are already setting labels, which is easy.

The dev rule is: "If you check a pull request to be mergeable, write a simple "+1" comment to it." and for good reason, a comment is likely to be preserved by GitHub and is clearly visible, unlike an emoticon clicklist. Core devs should continue to do this even if they also use the emoticons.

**Comment by paramat** [comment]
So anyway, to not conflict with automation, i change my request to:

Please also add the label and add a comment of support, whether or not you also use emoticons in the first post.

**Comment by Ezhh** [comment]
I don't really think they should be automated. 

At the moment we need two approvals for a merge, and these combine approval of the concept and technical parts of a PR (plus anything else that might matter). The problem is that everything needed for approval is delayed to the end, so someone can do all the work only to then have it rejected based on the concept, or perhaps worse still - they'll never try to contribute to begin with because they don't believe their idea will get support.

Of course contributors can already ask the devs for advance opinions before they start something, but there is no structure to that so many won't, and when they do there is no clear record it happened.

It's about finding a balance between being able to provide contributors with some reassurance that they are using their time well, and the amount of extra dev work needed. If you need to concept approve everything in advance, even  if limiting it to feature requests, it can easily be too much. Something shouldn't need concept approval for people to work on it, but it should be a way for devs to flag things they are very keen to see worked on and a thing people can request before starting work on something.

Where concept approval is given, it's simply moving forward a small part of the approval that is otherwise all given at the end (which is why two is still the right number). It doesn't guarantee a merge by any means, but should give a good chance that a PR won't be turned down based on the very idea of what it is (unless circumstances have changed in a way that forces reassessment).

**Comment by Wuzzy2** [comment]
I always thought it is a little bit silly to use the emoticons to count official support, or even make polls. That's not what they're meant for!
As paramat said, emoticons are mainly a *playful* thing and should be viewed as such.
I think paramat is spot on here.

**Comment by Ezhh** [comment]
The first post emoticons tallies are useful for a general idea of people's reactions, which is worth consideration when adding or changing a feature, but I wouldn't consider their use official support and didn't think they were ever meant to be used that way.

Approval of a concept, just like actual approval of a PR, should be indicated clearly with a distinct comment giving that approval and ideally any associated conditions that need to be met for it to hold.

**Comment by paramat** [comment]
After chatting to Ezhh, i agree that concept approvals need to be very specific and detailed about the implementation, for obvious reasons. So they are very different from the core dev support of an undetailed feature request, which is what this issue refers to.
So this issue is indeed independent from official concept approvals, my mistake for suggesting they could be combined, i'll edit my comments above.

The label is in use now, a GitHub issue was possibly not the best way to announce it, so closing. Concept approvals should be a separate issue.

**Comment by Ezhh** [comment]
Just as a thought, maybe info about how to know if an issue has dev support or something has concept approval should be documented in https://github.com/minetest/minetest/blob/master/CONTRIBUTING.md, or somewhere similar. (Assuming how to handle concept approvals gets agreed on at some point.)
